src/ledger-keyring.test.ts

```diff
@@ -230,12 +231,70 @@ describe('LedgerKeyring', function () {
     });
   });
 
+  describe('setDeviceId', function () {
+    it('should set the deviceId', function () {
+      keyring.setDeviceId('some-device');
+      expect(keyring.getDeviceId()).toBe('some-device');
+    });
+  });
+
+  describe('getDeviceId', function () {
+    it('should return the deviceId', function () {
+      keyring.setDeviceId('some-device');
+      expect(keyring.getDeviceId()).toBe('some-device');
+    });
+  });
+
+  describe('isConnected', function () {
+    it('should return true if bridge is connected', function () {
+      bridge.isDeviceConnected = true;
+      expect(keyring.isConnected()).toBe(true);
+    });
+
+    it('should return false if bridge is not connected', function () {
+      bridge.isDeviceConnected = false;
+      expect(keyring.isConnected()).toBe(false);
+    });
+  });
+
+  describe('getName', function () {
+    it('should return the name', function () {
+      expect(keyring.getName()).toBe('Ledger Hardware');
+    });
+  });
+
   describe('isUnlocked', function () {
     it('should return true if we have a public key', function () {
       expect(keyring.isUnlocked()).toBe(true);
     });
   });
 
+  describe('setAccountToUnlock', function () {
+    it('should set unlockedAccount', function () {
+      keyring.setAccountToUnlock(3);
+      expect(keyring.unlockedAccount).toBe(3);
+    });
+
+    it('should set unlockedAccount to 0 if argument is not number type', function () {
+      keyring.setAccountToUnlock('3' as unknown as number);
+      expect(keyring.unlockedAccount).toBe(3);
```

Comment:
The test says it should be 0 but here we're expecting it to be 3, is that correct? At a closer inspection, the `setAccountToUnlock` accepts a `string` and tries to `parseInt` it. Maybe we should improve that method a bit like this,

```ts
setAccountToUnlock(index: number) {
  if (typeof index !== 'number') {
    throw new Error('index must be a number');
  }

  this.unlockedAccount = index;
}
```

and then the test could be,

```ts
it('throws an error if index argument is not number type', function () {
  // @ts-expect-error we want to force an invalid type error
  expect(() => keyring.setAccountToUnlock('not a number')).toThrow('index must be a number');
});
```

Reply:
hi, regarding the code you suggest
```
setAccountToUnlock(index: number) {
  if (typeof index !== 'number') {
    throw new Error('index must be a number');
  }

  this.unlockedAccount = index;
}
```
Because index is already defined as number, the first if statement will always be false in typescript. this will lead to typescript lint warning.  i have simplified this setAccountToUnlock code as following
```
  setAccountToUnlock(index: number) {
    this.unlockedAccount = index;
  }
 ``` 
 above code will bypass typescript lint warning. 
 
 After change the above method signature to force index: number, the unit test will never needed anymore because typescript will enforce index to be number. 
 
 I have completed remove the unit tests for not number type check.